lib/plugins/aws/package/lib/generateCoreTemplate.test.js

```diff
@@ -45,6 +45,36 @@ describe('#generateCoreTemplate()', () => {
     };
   });
 
+  it('should reject non-HTTPS requests to the deployment bucket', () => {
+    return expect(awsPlugin.generateCoreTemplate()).to.be.fulfilled.then(() => {
+      const serverlessDeploymentBucketPolicy =
+        awsPlugin.serverless.service.provider.compiledCloudFormationTemplate.Resources
+          .ServerlessDeploymentBucketPolicy;
+
+      expect(serverlessDeploymentBucketPolicy).to.exist;
+      expect(serverlessDeploymentBucketPolicy.Type).to.equal('AWS::S3::BucketPolicy');
+      expect(serverlessDeploymentBucketPolicy.Properties).to.exist;
+      expect(serverlessDeploymentBucketPolicy.Properties.Bucket).to.deep.equal({
+        Ref: 'ServerlessDeploymentBucket',
+      });
+
+      expect(serverlessDeploymentBucketPolicy.Properties.PolicyDocument).to.exist;
+      expect(serverlessDeploymentBucketPolicy.Properties.PolicyDocument.Statement).to.exist;
+
+      expect(serverlessDeploymentBucketPolicy.Properties.PolicyDocument.Statement).to.deep.include({
+        Action: 's3:*',
+        Effect: 'Deny',
+        Principal: '*',
+        Resource: [
+          { 'Fn::Join': ['', ['arn:aws:s3:::', { Ref: 'ServerlessDeploymentBucket' }, '/*']] },
+        ],
+        Condition: {
+          Bool: { 'aws:SecureTransport': false },
+        },
+      });
```

Comment:
Doing it this way should prevent this test from breaking if additional statements are added in the future.